connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -371,6 +373,14 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
   private[connect] def listListenerIds(): Seq[String] = {
     listenerCache.keySet().asScala.toSeq
   }
+
+  /**
+   * An accumulator for Python executors.
+   *
+   * The accumulated results will be sent to the Python client via observed_metrics message.
+   */
+  private[connect] val pythonAccumulator: Option[PythonAccumulator] =
+    Try(session.sparkContext.collectionAccumulator[Array[Byte]]).toOption
```

Comment:
Hm, looks like we don't need `Try(...)` here? I took a cursory look, and seems it won't throw an exception.

Reply:
BTW, if the profile is disabled, we shouldn't probably create this accumulator to avoid performance issue.

Reply:
> looks like we don't need `Try(...)` here? 

In some tests, mocks of `session` or `sparkContext` are used and it throws an exception when creating accumulators.

Reply:
> if the profile is disabled, we shouldn't probably create this accumulator to avoid performance issue.

It needs to always have the accumulator because:
- it can't know whether or not / when the profiler is enabled
- to support the registered UDFs

What kind of performance issue do you concern?

Reply:
My concern is that regisgerting too many acumulators because calling this will create and register accumator for each session. Especially for Spark Connent, there could be a lot of Spark sessions

Reply:
There are already much more accumulators registered for each query, as `SQLMetrics`. I don't think one more accumulator per session could be an issue.

Reply:
👌